dot/parachain/collator-protocol/message_test.go

```diff
@@ -172,3 +178,226 @@ func TestDecodeCollationHandshake(t *testing.T) {
+				}, peerID)
+				return net
+			}(),
+		},
+		{
+			description: "fail if collator signature could not be verified",
+			declareMsg: Declare{
+				CollatorId:        collatorID.AsBytes(),
+				ParaID:            uint32(5),
+				CollatorSignature: parachaintypes.CollatorSignature{},
+			},
+			peerData: map[peer.ID]PeerData{
+				peerID: {
+					view: View{},
+					state: PeerStateInfo{
+						PeerState: Connected,
+					},
+				},
+			},
+			errString: "verifying signature",
+		},
+		{
+			description: "fail if collator signature is invalid and report the sender",
+			declareMsg: Declare{
+				CollatorId:        collatorID.AsBytes(),
+				ParaID:            uint32(5),
+				CollatorSignature: invalidCollatorSignature,
+			},
+			peerData: map[peer.ID]PeerData{
+				peerID: {
+					view: View{},
+					state: PeerStateInfo{
+						PeerState: Connected,
+					},
+				},
+			},
+			net: func() Network {
+				ctrl := gomock.NewController(t)
+				net := NewMockNetwork(ctrl)
+				net.EXPECT().ReportPeer(peerset.ReputationChange{
+					Value:  peerset.InvalidSignatureValue,
+					Reason: peerset.InvalidSignatureReason,
+				}, peerID)
+				return net
+			}(),
+			errString: crypto.ErrSignatureVerificationFailed.Error(),
+		},
+		{
+			// TODO: test that we disconnect sender in this case, after we add that functionality
+			description: "fail if paraID in Declare message is not assigned to our peer and report the sender",
+			declareMsg: Declare{
+				CollatorId:        collatorID.AsBytes(),
+				ParaID:            uint32(5),
+				CollatorSignature: collatorSignature,
+			},
+			peerData: map[peer.ID]PeerData{
+				peerID: {
+					view: View{},
+					state: PeerStateInfo{
+						PeerState: Connected,
+					},
+				},
+			},
+			currentAssignments: make(map[parachaintypes.ParaID]uint),
+			net: func() Network {
+				ctrl := gomock.NewController(t)
+				net := NewMockNetwork(ctrl)
+				net.EXPECT().ReportPeer(peerset.ReputationChange{
+					Value:  peerset.UnneededCollatorValue,
+					Reason: peerset.UnneededCollatorReason,
+				}, peerID)
+				return net
+			}(),
+		},
+		{
+			description: "success case: check if PeerState of the sender has changed to Collating from Connected",
+			declareMsg: Declare{
+				CollatorId:        collatorID.AsBytes(),
+				ParaID:            uint32(5),
+				CollatorSignature: collatorSignature,
+			},
+			peerData: map[peer.ID]PeerData{
+				peerID: {
+					view: View{},
+					state: PeerStateInfo{
+						PeerState: Connected,
+					},
+				},
+			},
+			currentAssignments: map[parachaintypes.ParaID]uint{
+				parachaintypes.ParaID(5): 1,
+			},
+			success: true,
+		},
+	}
+
+	for _, c := range testCases {
+		c := c
+		t.Run(c.description, func(t *testing.T) {
+			t.Parallel()
+			cpvs := CollatorProtocolValidatorSide{
+				net:                c.net,
+				peerData:           c.peerData,
+				currentAssignments: c.currentAssignments,
+			}
+			msg := NewCollationProtocol()
+			vdt_child := NewCollatorProtocolMessage()
```

Comment:
Use camel case instead
```suggestion
			vdtChild := NewCollatorProtocolMessage()
```